src/Cpp/1-getting-started/14-HelloCOM/HelloCOM.cpp

```diff
@@ -0,0 +1,330 @@
+#endif
+    _deviceContext.Reset();
+    _device.Reset();
+    Application::Cleanup();
+}
+
+bool HelloCOMApplication::Initialize()
+{
+    // This section initializes GLFW and creates a Window
+    if (!Application::Initialize())
+    {
+        return false;
+    }
+    glfwSetWindowUserPointer(GetWindow(), this);
+    glfwSetFramebufferSizeCallback(GetWindow(), OnResize);
+    const HWND nativeWindow = glfwGetWin32Window(GetWindow());
+    // This section initializes DirectX's devices and SwapChain
+    if (FAILED(CreateDXGIFactory1(__uuidof(IDXGIFactory1), &_dxgiFactory)))
+    {
+        std::cout << "DXGI: Unable to create DXGIFactory\n";
+        return false;
+    }
+    _dxgiFactory->MakeWindowAssociation(nativeWindow, 0);
+    constexpr D3D_FEATURE_LEVEL deviceFeatureLevel = D3D_FEATURE_LEVEL_11_0;
+    UINT deviceFlags = D3D11_CREATE_DEVICE_BGRA_SUPPORT;
+#if !defined(NDEBUG)
+    deviceFlags |= D3D11_CREATE_DEVICE_DEBUG;
+#endif
+    if (FAILED(D3D11CreateDevice(
+        nullptr,
+        D3D_DRIVER_TYPE_HARDWARE,
+        nullptr,
+        deviceFlags,
+        &deviceFeatureLevel,
+        1,
+        D3D11_SDK_VERSION,
+        &_device,
+        nullptr,
+        &_deviceContext)))
+    {
+        std::cout << "D3D11: Failed to create Device and Device Context\n";
+        return false;
+    }
+#if !defined(NDEBUG)
+    if (FAILED(_device.As(&_debug)))
+    {
+        std::cout << "D3D11: Failed to get Device Debug Context\n";
+        return false;
+    }
+    if (FAILED(_debug.As(&_debugInfoQueue)))
+    {
+        std::cout << "D3D11: Failed to get Debug Info Queue\n";
+        return false;
+    }
+    _debugInfoQueue->SetBreakOnSeverity(D3D11_MESSAGE_SEVERITY_CORRUPTION, true);
+    _debugInfoQueue->SetBreakOnSeverity(D3D11_MESSAGE_SEVERITY_ERROR, true);
+
+    D3D11_MESSAGE_ID hide[] =
+    {
+        D3D11_MESSAGE_ID_SETPRIVATEDATA_CHANGINGPARAMS,
+        // Add more message IDs here as needed
+    };
+
+    D3D11_INFO_QUEUE_FILTER filter = {};
+    filter.DenyList.NumIDs = _countof(hide);
+    filter.DenyList.pIDList = hide;
+    _debugInfoQueue->AddStorageFilterEntries(&filter);
+    _debugInfoQueue.Reset();
+#endif
+
+    DXGI_SWAP_CHAIN_DESC swapchainInfo;
+    swapchainInfo.BufferDesc.Width = _width;
+    swapchainInfo.BufferDesc.Height = _height;
+    swapchainInfo.BufferDesc.RefreshRate.Numerator = 0;
+    swapchainInfo.BufferDesc.RefreshRate.Denominator = 1;
+    swapchainInfo.BufferDesc.Format = DXGI_FORMAT_B8G8R8A8_UNORM_SRGB;
+    swapchainInfo.BufferDesc.ScanlineOrdering = DXGI_MODE_SCANLINE_ORDER_UNSPECIFIED;
+    swapchainInfo.BufferDesc.Scaling = DXGI_MODE_SCALING_STRETCHED;
+    swapchainInfo.SampleDesc.Count = 1;
+    swapchainInfo.SampleDesc.Quality = 0;
+    swapchainInfo.BufferUsage = DXGI_USAGE_RENDER_TARGET_OUTPUT;
+    swapchainInfo.BufferCount = 2;
+    swapchainInfo.OutputWindow = nativeWindow;
+    swapchainInfo.Windowed = true;
+    swapchainInfo.SwapEffect = DXGI_SWAP_EFFECT_DISCARD;
```

Comment:
I think we should also switch to the FLIP model, since we target mainly win10.

Reply:
FLIP model doesnt support SRGB output formats, only`DXGI_FORMAT_R16G16B16A16_FLOAT, DXGI_FORMAT_B8G8R8A8_UNORM, DXGI_FORMAT_R8G8B8A8_UNORM, DXGI_FORMAT_R10G10B10A2_UNORM`